python/pyarrow/_dataset.pyx

```diff
@@ -239,15 +239,17 @@ cdef class Dataset(_Weakrefable):
         ----------
         columns : list of str, default None
             The columns to project. This can be a list of column names to
-            include (order and duplicates will be preserved), or a dictionary
+            include (order and duplicates will be preserved) which may contain the
+            augmented fields such as `batch_index`, `fragment_index`, 
+            `last_in_fragment` and `filename`, or a dictionary
```

Comment:
A few things: the augmented fields start with a double underscore, right? Also, the wording is a little confusing, it makes it sound like we can only use the augmented fields in the list. Maybe we can put the list of augmented fields at the bottom or after this paragraph, e.g. "The list of columns or expressions may use the special fields __batch_index (the index of the batch within the fragment), ...., or __filename (the name of the source file or a description of the source fragment)."

Reply:
Agreed.  It would be good to change the wording here.

Reply:
Made the change, does this sound right now?